Wrappers/Python/ccpi/viewer/utils/io.py

```diff
@@ -42,3 +66,765 @@ def SaveRenderToPNG(render_window, filename):
+        '''
+        self._HDF5Compression = compression
+
+    def Write(self):
+        # check file ext
+        writer = self._GetWriter()
+        writer.Write()
+
+    def _GetWriter(self):
+        file_name = os.path.splitext(self._FileName)[0]
+
+        if self._FileFormat in ['nxs', 'h5', 'hdf5', '']:
+            self._FileName = file_name + '.hdf5'
+            writer = self._GetHDF5Writer()
+
+        elif self._FileFormat in ['mha']:
+            self._FileName = file_name + '.mha'
+            writer = self._GetMetaImageWriter()
+
+        else:
+            raise Exception("File format is not supported. Supported types include hdf5/nexus.")
+
+        writer.SetFileName(self._FileName)
+        return writer
+
+    def _GetHDF5Writer(self):
+        writer = cilviewerHDF5Writer()
+        writer.SetOriginalDataset(None, self._OriginalDatasetAttributes)
+        writer.SetChunking(self._Chunking)
+        writer.SetChunkShape(self._ChunkShape)
+        writer.SetHDF5Compression(self._HDF5Compression)
+        for i in range(0, len(self._ChildDatasets)):
+            writer.AddChildDataset(self._ChildDatasets[i], self._ChildDatasetsAttributes[i])
+        return writer
+
+    def _GetMetaImageWriter(self):
+        writer = vtk.vtkMetaImageWriter()
+        writer.SetInputData(self._ChildDatasets[0])
+        return writer
+
+
+class cilviewerHDF5Writer(ImageWriter):
+    '''
+    Expects to be writing an original dataset or attributes of the original dataset,
+    plus one or more 'child' versions of the dataset which have been resampled and/or cropped.
+    '''
+
+    def __init__(self):
+        super(cilviewerHDF5Writer, self).__init__()
+
+    def _ValidateChildDatasetAttributes(self, child_dataset, attributes):
+        if not isinstance(attributes, dict):
+            raise Exception("'attributes must be a dictionary.")
+        # check origin and spacing are set
+        # check resampling and cropping are set
+        child_attributes_schema = Schema({
+            'resampled': bool,
+            'cropped': bool,
+            'spacing': Or(list, tuple),
+            'origin': Or(list, tuple),
+            Optional('resample_z'): bool,
+            Optional(str): object  # allow any other keys and values
+        })
+        child_attributes_schema.validate(attributes)
+
+    def Write(self):
+        for var in [self._FileName, self._OriginalDatasetAttributes]:
+            if var is None:
+                raise Exception("file_name, dataset(/s) and attribute(/s), are required.")
+        for var in [self._ChildDatasets, self._ChildDatasetsAttributes]:
+            if var is []:
+                raise Exception("child dataset(/s) and attribute(/s), are required.")
+
+        with h5py.File(self._FileName, 'w') as f:
+
+            # give the file some important attributes
+            f.attrs['file_name'] = self._FileName
+            f.attrs['viewer_version'] = version
```

Comment:
The reason the docker github action is failing is because in io we are importing version, and saving it to the file we create here. for now we could not do this and it would stop the action from failing? I don't know how to fix the action.

Reply:
@paskino, shall I just not print the version here for now so that the action passes?

Reply:
That's what I've done for now, will make a linked issue later